contributing_to_docs/doc_guidelines.adoc

```diff
@@ -911,22 +911,134 @@ kind: Pod
 metadata:
   labels:
     test: liveness
-...
+# ...
 ----
 +
 or
 +
 ----
-...
+# ...
 spec:
   containers:
   - args:
     image: k8s.gcr.io/liveness
-...
+# ...
 ----
 +
 Do not use `[...]`, `<snip>`, or any other variant.
 
+* Do not use `[source,yaml]` for the following:
+
+** Bash "here" documents, such as the following example:
++
+....
+[source,terminal]
+----
+$ cat <<EOF| oc create -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mlistener
+  labels:
+    app: multicast-verify
+EOF
+----
+....
+
+** The output of `oc describe <pural> <object>`, such as in the following example:
++
+....
+[source,text]
+----
+Name:               node1.example.com
+Roles:              worker
+Labels:             beta.kubernetes.io/arch=amd64
+...
+Annotations:        cluster.k8s.io/machine: openshift-machine-api/ahardin-worker-us-east-2a-q5dzc
+...
+CreationTimestamp:  Wed, 13 Feb 2019 11:05:57 -0500
+----
+....
+
+
+=== YAML formatting for Kubernetes and OpenShift API objects
+The following formatting guidelines apply to YAML manifests, but do not apply to the installation configuration YAML specified by `install-config.yaml`.
+
+==== Required fields
+
+- Always include the `apiVersion` and `kind` so that a user always knows the context of the YAML.
+- Always include the full hierarchy to a deeply nested key.
+- For objects that are in the global scope, such as for `config.openshift.io` API group, always include the `metadata.name` for the object, which is usually `cluster`.
+
+.Example API object in the global scope
+----
+apiVersion: config.openshift.io/v1
+kind: Scheduler
+metadata:
+  name: cluster
+# ...
+spec:
+  defaultNodeSelector: node-role.kubernetes.io/app=
+# ...
+----
+
+.Example deeply nested key with full context for `.ports` array
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod1
+  namespace: default
+spec:
+  containers:
+  - name: web
+    image: nginx
+    ports:
+    - name: web
+      containerPort: 80
+      protocol: TCP
+----
+
+==== Formatting
+Use the following conventions govern the layout of YAML for API objects.
+
+- Always begin YAML flush against the left margin
+- Always use 2 space indentation
+- Always use short array indentation
+- Always include a space immediate after the colon for keys
+- Always use block style for complex strings, such as embedded JSON or text blocks
+- When truncating YAML, always precede an ellipsis with a pound so that the YAML is valid.
+
```

Comment:
Maybe it doesn't need to be said, but I'm wondering if it would be worthwhile to add a bullet that says something about, if possible, test your YAML example in a test cluster to make sure that it is valid.